api/client/export.go

```diff
@@ -40,6 +38,33 @@ func (cli *DockerCli) CmdExport(args ...string) error {
 	}
 	defer responseBody.Close()
 
+	var tmpFile *os.File
+	if *outfile != "" {
+		tmpFile, err = ioutil.TempFile(filepath.Dir(*outfile), ".docker_export_")
+		if err != nil {
+			return err
+		}
+		output = tmpFile
+	}
 	_, err = io.Copy(output, responseBody)
-	return err
+
+	if err != nil {
+		if *outfile != "" {
+			tmpPath := tmpFile.Name()
+			tmpFile.Close()
+			os.Remove(tmpPath)
```

Comment:
@jfrazelle maybe in this way? is call File.Close() two times ok?
https://gist.github.com/daehyeok/77319f3ca7025ac3a2cb